nemo/core/neural_types/elements.py

```diff
@@ -0,0 +1,191 @@
+        """Override this property to parametrize your type"""
+        return {}
+
+    @property
+    def fields(self) -> Optional[Tuple]:
+        return None
+
+    def compare(self, second) -> NeuralTypeComparisonResult:
+        # First, check general compatibility
+        first_t = type(self)
+        second_t = type(second)
+
+        if first_t == second_t:
+            result = NeuralTypeComparisonResult.SAME
+        elif issubclass(first_t, second_t):
+            result = NeuralTypeComparisonResult.LESS
+        elif issubclass(second_t, first_t):
+            result = NeuralTypeComparisonResult.GREATER
+        else:
+            result = NeuralTypeComparisonResult.INCOMPATIBLE
+
+        if result != NeuralTypeComparisonResult.SAME:
+            return result
+        else:
+            # now check that all parameters match
+            check_params = set(self.type_parameters.keys()) == set(second.type_parameters.keys())
+            if check_params is False:
+                return NeuralTypeComparisonResult.SAME_TYPE_INCOMPATIBLE_PARAMS
+            else:
+                for k1, v1 in self.type_parameters.items():
+                    if v1 != second.type_parameters[k1]:
+                        return NeuralTypeComparisonResult.SAME_TYPE_INCOMPATIBLE_PARAMS
+            # check that all fields match
+            if self.fields == second.fields:
+                return NeuralTypeComparisonResult.SAME
+            else:
+                return NeuralTypeComparisonResult.INCOMPATIBLE
+
+
+class VoidType(ElementType):
+    """Void-like type which is compatible with everything
+    """
+
+    def __str__(self):
+        return str("void type. compatible with everything")
+
+    def compare(cls, second: abc.ABCMeta) -> NeuralTypeComparisonResult:
+        return NeuralTypeComparisonResult.SAME
+
+
+# TODO: Consider moving these files elsewhere
+class ChannelType(ElementType):
+    def __str__(self):
+        return "convolutional channel value"
+
+
+class EmbeddedTextType(ChannelType):
+    def __str__(self):
+        return "text embedding"
+
+
+class LogitsType(ElementType):
+    def __str__(self):
+        return "neural type representing logits"
+
+
+class LogprobsType(ElementType):
+    def __str__(self):
+        return "neural type representing log probabilities"
+
+
+class LabelsType(ElementType):
+    def __str__(self):
+        return "neural type representing labels"
+
+
+class LengthsType(ElementType):
+    def __str__(self):
+        return "neural type representing lengths of something"
+
+
+class LossType(ElementType):
+    def __str__(self):
+        return "neural type representing loss value"
+
+
+class EncodedRepresentation(ChannelType):
+    def __str__(self):
+        return "encoded representation, for example, encoder's output"
+
+
+class AcousticEncodedRepresentation(EncodedRepresentation):
+    def __str__(self):
+        return "encoded representation returned by the acoustic encoder model"
```

Comment:
Description of classes are missing - as a result we will have empty documentation sections.

Reply:
will fix